www/src/pages/table/OntSearchComponent.tsx

```diff
@@ -41,59 +46,132 @@ function useSearchComponent(
+    value: ``,
+  };
 
-    if (Array.isArray(props.value)) {
-      // leverage the fact that the three stores are not re-fetched
-      if (!store.refetch) {
-        store.fetchByIds(props.value);
-      } else {
-        store.fetchByIds(props.value.filter((id) => !store.records.has(id)));
-      }
-    } else if (store.get(props.value) === undefined) {
-      store.fetchById(props.value);
+  // search for additional values if it's not in the list
+  const onSearch = (query: string) => {
+    if (query === "") {
+      return;
     }
-  }, [props.value]);
+    const searchResults: SearchOptions[] = [];
+    loaderOption.filterlabel = query;
 
-  // gather all options already in the store, leverage the fact
-  // that property store is readonly
-  const options = useMemo(() => {
-    const options = [];
-    for (const r of store.iter()) {
-      options.push({
-        value: r.id,
-        label: r.readableLabel,
-      });
+    setSearchOptions([loaderOption]);
+    if (storeName === "classStore") {
+      store
+        .fetchSearchResults(query)
+        .then((data) => {
+          data.forEach((searchResult: ClassTextSearchResult) => {
+            searchResults.push({
+              type: "class",
+              id: searchResult.id,
+              label: (
+                <LabelComponent
+                  id={searchResult.id}
+                  description={searchResult.description}
+                  label={searchResult.label}
+                />
+              ),
+              filterlabel: `${searchResult.label} (${searchResult.id})`,
+              value: `${searchResult.id}`,
+            });
+          });
+          return searchResults;
+        })
+        .then((searchResults) => {
+          setSearchOptions([...searchResults]);
+        })
+        .catch(function (error: any) {
+          console.error(error);
+        });
+    } else if (storeName === "propertyStore") {
+      store
+        .fetchSearchResults(query)
+        .then((data) => {
+          data.forEach((searchResult: PropertyTextSearchResult) => {
+            searchResults.push({
+              id: searchResult.id,
+              label: (
+                <LabelComponent
+                  id={searchResult.id}
+                  label={searchResult.label}
+                  description={searchResult.description}
+                />
+              ),
+              filterlabel: `${searchResult.label} (${searchResult.id})`,
+              value: `${searchResult.id}`,
+            });
+          });
+          return searchResults;
+        })
+        .then((searchResults) => {
+          setSearchOptions([...searchResults]);
+        })
+        .catch(function (error: any) {
+          console.error(error);
+        });
+    } else {
+      store
+        .fetchSearchResults(query)
+        .then((data) => {
+          data.forEach((searchResult: EntityTextSearchResult) => {
+            searchResults.push({
+              id: searchResult.id,
```

Comment:
duplicated code, you should simplify them.

Reply:
Fixed it